Scheduler/Controllers/GoalController.cs

```diff
@@ -1,89 +1,117 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
 
 namespace Scheduler.Controllers
 {
 	[Route("api/[controller]")]
 	[ApiController]
 	public class GoalController : ControllerBase
 	{
-		[HttpPost]
+		private readonly GoalContext _goalContext;
+
+		public GoalController(GoalContext goalContext)
+		{
+			_goalContext = goalContext;
+		}
+
+		#region Goal CRUD operations
+		[Route("api/[controller]/[action]")]
+		[HttpPost("AddGoal")]
 		public long AddGoal(Goal goal)
 		{
-			using (var goalContext = new GoalContext())
-			{
-				goalContext.Add(goal);
-				goalContext.SaveChanges();
-				return goal.GoalId;
-			}
+			goal.GoalId = 0;
+			_goalContext.Add(goal);
+			_goalContext.SaveChanges();
+
+			return goal.GoalId;
 		}
 
-		[HttpPut]
+		[Route("api/[controller]/[action]")]
+		[HttpPut("UpdateGoal")]
 		public void UpdateGoal(Goal goal)
 		{
-			using (var goalContext = new GoalContext())
+			var currentGoal = _goalContext.Goals.Find(goal.GoalId);
+
+			if (currentGoal != null)
 			{
-				goalContext.Update(goal);
-				goalContext.SaveChanges();
+				_goalContext.Entry(currentGoal).CurrentValues.SetValues(goal);
+				_goalContext.SaveChanges();
 			}
 		}
 
-		[HttpDelete]
+		[Route("api/[controller]/[action]")]
+		[HttpDelete("DeleteGoal")]
 		public void DeleteGoal(Goal goal)
 		{
-			using (var goalContext = new GoalContext())
+			var currentGoal = _goalContext.Goals.Find(goal.GoalId);
+
+			if (currentGoal != null)
 			{
-				goalContext.Remove(goal);
-				goalContext.SaveChanges();
+				_goalContext.Remove(currentGoal);
+				_goalContext.SaveChanges();
 			}
 		}
 
-		[HttpGet]
+		[Route("api/[controller]/[action]")]
+		[HttpGet("GetGoalById")]
 		public Goal GetGoalById(long goalId)
 		{
-			using (var goalContext = new GoalContext())
-			{
-				return goalContext.Goals.Find(goalId);
-			}
+			return _goalContext.Goals.Find(goalId);
 		}
 
-		[HttpPost]
+		[Route("api/[controller]/[action]")]
+		[HttpGet("GetAllGoals")]
+		public IEnumerable<Goal> GetAllGoals()
+		{
+			return _goalContext.Goals;
+		}
+		#endregion
+
+		#region Notification CRUD operations
+		[Route("api/[controller]/[action]")]
+		[HttpPost("AddNotification")]
 		public long AddNotification(Notification notification)
```

Comment:
We can't add notifications without goal. So we need to pass goalid.
I suggest to use url in one of this formats:
`api/goal/1/addnotification  `
or
`api/goal(1)/addnotification`